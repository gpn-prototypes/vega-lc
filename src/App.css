html,
body,
#root {
  margin: 0;
  padding: 0;

  width: 100%;
  height: 100%;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  width: 100%;
  height: 100%;

  background: var(--color-bg-default);
}
