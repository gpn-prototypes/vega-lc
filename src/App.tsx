import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { presetGpnDark, Theme } from '@gpn-prototypes/vega-ui';

import { GeologicalExploration } from './components/geological-exploration';
import Navigation from './components/navigation';

import './App.css';
import './styles/colors.css';

export const App = (): React.ReactElement => {
  return (
    <Theme className="App" preset={presetGpnDark}>
      <Navigation />
      <Switch>
        <Route exact component={GeologicalExploration} path="/" />
      </Switch>
    </Theme>
  );
};
