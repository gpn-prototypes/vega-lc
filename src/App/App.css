:root {
  --tooltip-z-index: 99;
}

* {
  box-sizing: border-box;
}

html,
body,
#root {
  margin: 0;
  padding: 0;

  width: 100%;
  height: 100%;
}

.LC__App-Wrapper {
  z-index: 0;

  display: flex;
  flex-direction: column;
  flex-grow: 1;

  width: 100%;
  height: 100%;
}

.LC__App {
  display: flex;
  overflow: hidden;
  flex-direction: column;
  flex-grow: 1;

  min-height: 100%;

  color: var(--color-typo-primary);
  background-color: var(--color-bg-default);
}

.Tooltip {
  z-index: var(--tooltip-z-index);
}

.SelectDropdown {
  z-index: 1;
}
