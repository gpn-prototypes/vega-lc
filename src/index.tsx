import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { ApolloProvider } from '@apollo/client';
import { Root } from '@gpn-prototypes/vega-root';

import getStore from './redux-store/index';
import { App } from './App';
import client from './client';

import './App.css';

ReactDOM.render(
  <Root initialPortals={[{ name: 'modalRoot' }]} defaultTheme="dark">
    <ApolloProvider client={client}>
      <Provider store={getStore()}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    </ApolloProvider>
  </Root>,
  document.getElementById('root'),
);
