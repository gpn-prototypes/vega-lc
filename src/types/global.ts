import ResizeObserver from 'resize-observer-polyfill';

declare global {
  interface Window {
    ResizeObserver: typeof ResizeObserver;
  }
}
