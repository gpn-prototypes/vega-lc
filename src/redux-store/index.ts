import { applyMiddleware, combineReducers, createStore, Store } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore } from 'redux-persist';
import thunkMiddleware from 'redux-thunk';

import activitiesReducer from './activities/reducer';
import groupObjectsReducer from './group-objects/reducer';
import logicConstructorReducer from './logic-constructor/reducer';
import notificationsReducer from './notifications/reducer';
import projectStructureReducer from './project-structure/reducer';
import versionReducer from './version/reducer';

import { StoreLC } from '@/types/redux-store';

const rootReducer = combineReducers({
  activities: activitiesReducer,
  groupObjects: groupObjectsReducer,
  projectStructure: projectStructureReducer,
  logicConstructor: logicConstructorReducer,
  version: versionReducer,
  notifications: notificationsReducer,
});

const getStore = (initialState?: StoreLC): Store =>
  createStore(
    rootReducer,
    initialState,
    composeWithDevTools(applyMiddleware(thunkMiddleware.withExtraArgument({}))),
  );

export const store = getStore();

export const persistor = persistStore(store);
