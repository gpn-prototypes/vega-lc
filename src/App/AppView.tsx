import React, { useContext, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Loader } from '@gpn-prototypes/vega-ui';

import { cnApp } from './cn-app';

import './App.css';
import '../styles/colors.css';

import { GeologicalExploration } from '@/components/geological-exploration';
import { ProjectErrors } from '@/components/project-errors';
import { ProjectContext } from '@/react-context/providers';
import { fetchVersion } from '@/redux-store/version/actions';
import { StoreLC } from '@/types/redux-store';

export const AppView = (): React.ReactElement => {
  const dispatch = useDispatch();

  const { initialized } = useContext(ProjectContext);

  const isLoading = useSelector<StoreLC, boolean>(
    (state) => !state.projectStructure.projectStructureQuery,
  );

  useEffect(() => {
    if (initialized) {
      dispatch(fetchVersion());
    }
  }, [dispatch, initialized]);

  return (
    <div className={cnApp('App')}>
      {isLoading ? <Loader size="m" /> : <GeologicalExploration />}
      <ProjectErrors />
    </div>
  );
};
