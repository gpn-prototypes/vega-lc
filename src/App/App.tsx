import React from 'react';
import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
import { Root } from '@gpn-prototypes/vega-ui';

import { Providers } from '../components/providers';

import { AppView } from './AppView';
import classNames from './cn-app';

import './App.css';

interface AppProps {
  graphqlClient?: ApolloClient<NormalizedCacheObject>;
}

export const App: React.FC<AppProps> = (props) => {
  const { graphqlClient } = props;

  return (
    <Root
      initialPortals={[{ name: 'modalRoot' }]}
      defaultTheme="dark"
      className={classNames('App-Wrapper').toString()}
    >
      <Providers graphqlClient={graphqlClient}>
        <AppView />
      </Providers>
    </Root>
  );
};
