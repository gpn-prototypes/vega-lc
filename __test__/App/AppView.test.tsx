import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react';
import ResizeObserver from 'resize-observer-polyfill';

import '../../src/types/global';

import { AppView } from '../../src/App/AppView';
import { ProjectContext } from '../../src/react-context/providers';
import { store } from '../../src/redux-store';

// jest.mock('../../src/components/geological-exploration', () => {
//   return {
//     GeologicalExploration: () => {
//       return <div />;
//     },
//   };
// });

const mockClearStore = jest.fn();

jest.mock('../../src/redux-store/clear/actions', () => {
  return {
    clearStore: mockClearStore,
  };
});

beforeEach(() => {
  mockClearStore.mockClear();
});

const renderComponent = (projectId = '', initialized = false) => {
  return render(
    <Provider store={store}>
      <ProjectContext.Provider value={{ projectId, initialized }}>
        <AppView />
      </ProjectContext.Provider>
    </Provider>,
  );
};

describe('AppView', () => {
  test('рендерится без ошибок', () => {
    window.ResizeObserver = ResizeObserver;

    const dom = renderer
      .create(
        <Provider store={store}>
          <AppView />
        </Provider>,
      )
      .toJSON();
    expect(dom).toMatchSnapshot();
  });

  test('вызывается очистка store', async () => {
    await renderComponent();

    waitFor(() => expect(mockClearStore).toHaveBeenCalledTimes(1));
  });
});
