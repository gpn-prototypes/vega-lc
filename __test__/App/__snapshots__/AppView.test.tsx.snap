// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView рендерится без ошибок 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <div
    className="Loader Loader_size_m"
  >
    <div
      className="Loader-Dot"
    />
  </div>
  <div
    className="LC_Notifications"
  >
    <div
      className="SnackBar"
    />
  </div>
</div>
`;
